src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -823,25 +819,15 @@ internal EXPR BindToField(EXPR pOptionalObject, FieldWithType fwt, BindingFlag b
 
             EXPRFIELD pResult;
             {
-                bool isLValue = false;
-                if ((pOptionalObject != null && pOptionalObject.type.IsPointerType()) || objectIsLvalue(pOptionalObject))
-                {
-                    isLValue = true;
-                }
+                bool isLValue = pOptionalObject != null && pOptionalObject.type.IsPointerType() || objectIsLvalue(pOptionalObject);
+
                 // Exception: a readonly field is not an lvalue unless we're in the constructor/static constructor appropriate
                 // for the field.
-                if (RespectReadonly() && fwt.Field().isReadOnly)
-                {
-                    if (ContainingAgg() == null ||
-                        !InMethod() || !InConstructor() ||
-                        fwt.Field().getClass() != ContainingAgg() ||
-                        InStaticMethod() != fwt.Field().isStatic ||
-                        (pOptionalObject != null && !isThisPointer(pOptionalObject)) ||
-                        InAnonymousMethod())
-                    {
-                        isLValue = false;
-                    }
-                }
+                if (RespectReadonly() && fwt.Field().isReadOnly
+                    && (ContainingAgg() == null || !InMethod() || !InConstructor()
+                        || fwt.Field().getClass() != ContainingAgg() || InStaticMethod() != fwt.Field().isStatic
+                        || (pOptionalObject != null && !isThisPointer(pOptionalObject)) || InAnonymousMethod()))
+                    isLValue = false;
```

Comment:
I don't know that this change is better... it's harder to read, at least for me.